source/app-development/tutorials-interactive-apps/add-custom-queue/global-static-list.rst

```diff
@@ -0,0 +1,251 @@
+#. :ref:`app-development-tutorials-interactive-apps-add-custom-queue-global-static-list-add-custom-attribute`
+#. :ref:`app-development-tutorials-interactive-apps-add-custom-queue-global-static-list-handle-custom-attribute`
+
+.. _app-development-tutorials-interactive-apps-add-custom-queue-global-static-list-modify-cluster-configuration:
+
+Modify Cluster Configuration
+----------------------------
+
+.. danger::
+
+   This assumes you have walked through :ref:`add-cluster-config` for the
+   cluster you intend on submitting the Interactive App to.
+
+We will add a ``custom:`` field to our cluster configuration file if it doesn't
+already exist and then introduce a custom field (named anything your heart
+desires) with a list of available queues/partitions:
+
+.. code-block:: yaml
+   :emphasize-lines: 10-
+
+   # /etc/ood/config/clusters.d/my_cluster.yml
+   ---
+   v2:
+     metadata:
+       title: "My Cluster"
+     login:
+       host: "my_cluster.my_center.edu"
+     job:
+       # ... resource manager specific options here ...
+     custom:
+       queues: # this can be named anything your heart desires
+         - "queue1"
+         - "queue2"
+
+.. warning::
+
+   Be sure to click "Restart Web Server" on the Dashboard everytime you edit
+   the Cluster Configuration file for changes to take effect within the app.
+
+.. _app-development-tutorials-interactive-apps-add-custom-queue-global-static-list-add-custom-attribute:
+
+Add Custom Attribute to Form
+----------------------------
+
+We want to **replace** the ``bc_queue`` form attribute with a custom HTML
+``<select>`` element (a drop-down list of options) built up of queue/partition
+options read from the corresponding cluster configuration file.
+
+.. note::
+
+   You can read more about customizing attributes in the ``form.yml`` file for
+   Interactive Apps under the
+   :ref:`interactive-development-form-customizing-attributes` section.
+
+#. First we rename the ``form.yml`` for the Jupyter Interactive App to
+   ``form.yml.erb``:
+
+   .. code-block:: shell
+
+      mv ~/ondemand/dev/jupyter/form.yml ~/ondemand/dev/jupyter/form.yml.erb
+
+   This will cause the YAML file to be processed using the `eRuby (embedded
+   Ruby)`_ templating system, which allows us to embed Ruby_ code into the YAML
+   configuration file for flow control, variable substitution, and more.
+
+#. Next we start with the default ``form.yml`` for the Jupyter Interactive App:
+
+   .. code-block:: yaml
+
+      # ~/ondemand/dev/jupyter/form.yml.erb
+      ---
+      cluster: "my_cluster"
+      attributes:
+        modules: "python"
+        conda_extensions: "1"
+        extra_jupyter_args: ""
+      form:
+        - modules
+        - conda_extensions
+        - extra_jupyter_args
+        - bc_account
+        - bc_queue
+        - bc_num_hours
+        - bc_num_slots
+        - bc_email_on_started
+
+#. We remove the following line from this file:
+
+   .. code-block:: yaml
+      :emphasize-lines: 2
```

Comment:
What about using the diff formatting instead of just yaml and emphasize? In https://osc.github.io/ood-documentation-test/add-queue-dropdown/app-development/tutorials-passenger-apps/ps-to-quota.html#edit-to-run-and-parse-quota I split it into two blocks of code, showing the code diff, and then showing the final block with the code formatting. I probably could update https://osc.github.io/ood-documentation-test/add-queue-dropdown/app-development/tutorials-passenger-apps/ps-to-quota.html#edit-to-run-and-parse-quota to use emphasize-lines for the lines that were changed/modified like you do here though.

Reply:
I originally had diff, but it was a little confusing when I would glance at it because each of the lines had dashes already, but that one line had two dashes.

So it would look like:

> We remove the following line from this file:
> 
> ```diff
>   - bc_account
> - - bc_queue
>   - bc_num_hours
> ```

I am not married to either solution, I just went with emphasize because it was identical to the diff but didn't have that extra dash that "might" confuse someone and the difference in color (although I wish I could alter the emphasize color).

If you want me to go back to using diff I can.

Reply:
I think the diff only works well if you call out that you are showing a diff, and works much better if you show removal and addition i.e. see the top of this as an example:

https://osc.github.io/ood-documentation-test/add-queue-dropdown-alt/app-development/tutorials-interactive-apps/add-custom-queue/local-static-list.html

But I do see value in showing removing the field and reloading as a separate step, to see the field being removed. Regardless, instead of saying

> We remove the following line from this file:

I would be more explicit and say:

> We remove the bc_queue field from this file:

When I was reading through the plain text, I saw:

>We remove the following line from this file:
>
>```
>- bc_account
>- bc_queue
>- bc_num_hours
>```

And I asked "why are we removing bc_account, maybe that was a mistake".

Reply:
Also, I trust whatever you decide on this and don't need to review it again.